extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeSearchQueryHandlerFactory.java

```diff
@@ -72,4 +73,24 @@ public String getUrl(String searchString, List<String> contentFilters, String so
 //                MUSIC_ARTISTS
         };
     }
+
+    public static String getSearchParameter(final String contentFilter) {
+        if (isNullOrEmpty(contentFilter)) return "";
+        switch (contentFilter) {
+                case VIDEOS:
+                    return "EgIQAQ%3D%3D";
+                case CHANNELS:
+                    return "EgIQAg%3D%3D";
+                case PLAYLISTS:
+                    return "EgIQAw%3D%3D";
+                case ALL:
+                case MUSIC_SONGS:
+                case MUSIC_VIDEOS:
+                case MUSIC_ALBUMS:
+                case MUSIC_PLAYLISTS:
+                case MUSIC_ARTISTS:
```

Comment:
Are this unhandled?
Also, I think above here in getUrl(), you can replace:
```java
                    case VIDEOS:
                        return SEARCH_URL + URLEncoder.encode(searchString, UTF_8) + "&sp=EgIQAQ%253D%253D";
                    case CHANNELS:
                        return SEARCH_URL + URLEncoder.encode(searchString, UTF_8) + "&sp=EgIQAg%253D%253D";
                    case PLAYLISTS:
                        return SEARCH_URL + URLEncoder.encode(searchString, UTF_8) + "&sp=EgIQAw%253D%253D";
```
with
```java
                    case VIDEOS:
                    case CHANNELS:
                    case PLAYLISTS:
                        return SEARCH_URL + URLEncoder.encode(searchString, UTF_8) + "&sp=" + getSearchParameter(contentFilters.get(0));
```

Reply:
> Are this unhandled?

The getSearchParameter is only used by `YoutubeSearchQueryHandlerFactory` and `YoutubeSearchExtractor.onFetchPage` so I don't know what to do.

Reply:
> Also, I think above here in getUrl(), you can replace:

No, I can't because the params different:

- `EgIQAQ%253D%253D` vs `EgIQAQ%3D%3D` for videos;
- `EgIQAg%253D%253D` vs `EgIQAg%3D%3D` for channels;
- `EgIQAw%253D%253D` vs `EgIQAw%3D%3D` for playlists.

Reply:
Ok then, nevermind